### PR TITLE
refactoring the nested cases

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: lib/entice/web/client.ex [hunk 1]
```diff
@@ -91,26 +91,19 @@ defmodule Entice.Web.Client do
...
-                  :error -> {:ok, :offline, friend_name}
-                  player -> {:ok, client.online_status, player[Player.Name].name}
-                end
-            end
-        end
-      #If char with that name has been deleted (If name is then taken by another account it will be a problem)
-      _ -> {:ok, :offline, friend_name}
-    end
+    with {:ok, account_id}                   <- Queries.get_account_id(friend_name),
+         client_id when is_binary(client_id) <- Client.Server.get_client_by_account_id(account_id),
```
Comment (ephe-meral) [inline lib/entice/web/client.ex]: `%{} = client_id` for short :)

File: lib/entice/web/client.ex [hunk 2]
```diff
@@ -91,26 +91,19 @@ defmodule Entice.Web.Client do
...
+         {:ok, client}                       <- Entity.fetch_attribute(client_id, Clienty),
+         player when is_map(player)          <- Player.attributes(client.entity_id) do
```
Comment (ephe-meral) [inline lib/entice/web/client.ex]: `%{} = player` for short :)

File: lib/entice/web/client.ex [hunk 3]
```diff
@@ -91,26 +91,19 @@ defmodule Entice.Web.Client do
...
-                  player -> {:ok, client.online_status, player[Player.Name].name}
-                end
-            end
-        end
-      #If char with that name has been deleted (If name is then taken by another account it will be a problem)
-      _ -> {:ok, :offline, friend_name}
-    end
+    with {:ok, account_id}                   <- Queries.get_account_id(friend_name),
+         client_id when is_binary(client_id) <- Client.Server.get_client_by_account_id(account_id),
+         {:ok, client}                       <- Entity.fetch_attribute(client_id, Clienty),
```
Comment (ephe-meral) [inline lib/entice/web/client.ex]: Clienty is probably not correct :P

File: lib/entice/web/client.ex [hunk 4]
```diff
@@ -91,26 +91,19 @@ defmodule Entice.Web.Client do
...
+         player when is_map(player)          <- Player.attributes(client.entity_id) do
+
```
Comment (ephe-meral) [inline lib/entice/web/client.ex]: not needed, just have a 2 spaces indent for the next line and thats enough visual difference :D

File: lib/entice/web/client.ex [hunk 5]
```diff
@@ -91,26 +91,19 @@ defmodule Entice.Web.Client do
...
+         {:ok, client.online_status, player[Player.Name].name}
+    else
+      nil    -> {:ok, :offline, friend_name}
```
Comment (ephe-meral) [inline lib/entice/web/client.ex]: those two lines can be expressed in `_ -> {:ok, :offline ...`

File: lib/entice/web/client.ex [hunk 6]
```diff
@@ -95,13 +95,11 @@ defmodule Entice.Web.Client do
   def get_status(friend_name) do
     with {:ok, account_id}                   <- Queries.get_account_id(friend_name),
          client_id when is_binary(client_id) <- Client.Server.get_client_by_account_id(account_id),
-         {:ok, client}                       <- Entity.fetch_attribute(client_id, Clienty),
-         player when is_map(player)          <- Player.attributes(client.entity_id) do
-
-         {:ok, client.online_status, player[Player.Name].name}
+         {:ok, client}                       <- Entity.fetch_attribute(client_id, Client),
+         %{} = player                        <- Player.attributes(client.entity_id) do
+    {:ok, client.online_status, player[Player.Name].name}
```
Comment (ephe-meral) [inline lib/entice/web/client.ex]: +2 spaces ;) - we're in a block

**Comment by ephe-meral** [comment]
@Sireeni you'll need to update the travis file as well, otherwise the build will continue to fail ;)

**Comment by ghost** [comment]
Oh that's the reason! Travis is forever loading on my browser so I can't really see what's the problem :')
btw @ephe-meral do we need all the travis.yml hack to use elixir? they have native support for it now!

**Comment by ephe-meral** [comment]
:heart:

**Pull request merged**